dotnet/src/Functions/Functions.OpenAPI/Extensions/OpenApiSchemaExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Writers;
+
+namespace Microsoft.SemanticKernel.Functions.OpenAPI.Extensions;
+internal static class OpenApiSchemaExtensions
+{
+    /// <summary>
+    /// Gets a JSON serialized representation of an <see cref="OpenApiSchema"/>
+    /// </summary>
+    /// <param name="schema">The schema.</param>
+    /// <returns></returns>
+    internal static JsonDocument ToJsonDocument(this OpenApiSchema schema)
+    {
+        var schemaBuilder = new StringBuilder();
+        schema.SerializeAsV3WithoutReference(new OpenApiJsonWriter(new StringWriter(schemaBuilder)));
```

Comment:
If the schema is an array of objects, this doesn't seem to do as named.

> {
  "type": "array",
  "items": {
    "$ref": "#/components/schemas/Product"
  }
}

or
> {"title":"ProductResponse","type":"object","properties":{"products":{"type":"array","items":{"$ref":"#/components/schemas/Product"}}}}

Reply:
Hmm, let me look into this...

Reply:
I pushed up a change that appears to fix this, can you give it another shot?